cider/exec/nextgen/jitlib/base/JITFunction.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ * Copyright (c) OmniSci, Inc. and its affiliates.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#ifndef JITLIB_JIT_Function_H
+#define JITLIB_JIT_Function_H
+
+#include <boost/container/small_vector.hpp>
+
+#include "exec/nextgen/jitlib/base/ValueTypes.h"
+
+namespace jitlib {
+enum JITFunctionParamAttr : uint64_t {};
+
+struct JITFunctionParam {
+  const char* name{""};
```

Comment:
Yes, I prefer to name the function parameters as literal because the function signature and the semantics of parameters should be fixed at compile-time. If the parameters' name and others' name may changed at runtime, we should maintain a ```string``` for this. Any ideas?

Reply:
> Yes, I prefer to name the function parameters as literal because the function signature and the semantics of parameters should be fixed at compile-time. 

this should be ok